cms/src/main/resources/api-specification/codegen-templates/index.mustache

```diff
@@ -6,131 +6,102 @@
-                                                {{/partial}}
-                                                {{>params_table}}
-                                            {{/hasFormParams}}
-
-                                            {{#hasBodyParam}}
-                                                <div class="httpparams">
-                                                    <h5 class="nhsd-t-heading-s">Body</h5>
-                                                    {{#bodyParams}}
-                                                        {{#required}}<div class="nhsd-!t-col-red">Required</div>{{/required}}
-
-                                                        {{#vendorExtensions.x-body}}
-                                                            {{>body_param}}
-                                                        {{/vendorExtensions.x-body}}
-                                                    {{/bodyParams}}
-                                                </div>
-                                            {{/hasBodyParam}}
-
-                                            {{#responses}}
+                                    <!-- https://digital.nhs.uk/about-nhs-digital/corporate-information-and-documents/nhs-digital-style-guidelines/features-of-the-nhs-digital-website-cms/call-to-action-cta-buttons -->
+                                    <h3 id="api-{{baseName}}-{{nickname}}" class="nhsd-t-heading-l">{{summary}}</h3>
+
+                                    <div class="endpoint_path">
+                                        <div class="method">{{httpMethod}}</div>
+                                        <div class="pre"><pre>{{path}}</pre></div>
+                                        {{#ifVariableIsFalse "tryThisApiDisabled"}}
+                                            {{#eq baseName 'Default'}}{{#assign "sanitizedBaseName"}}{{lower baseName}}{{/assign}}{{else}}{{#assign "sanitizedBaseName"}}{{baseName}}{{/assign}}{{/eq}}
+                                                <div class="try"><a onclick="tryEndpointNow('/{{sanitizedBaseName}}/{{operationId}}')" class="ctabtn-left ctabtn--nhs-digital-button" onkeyup="return vjsu.onKeyUp(event)">Try this API</a></div>
+                                        {{/ifVariableIsFalse}}
+                                    </div>
+
+                                    {{{markdown notes levelToNormaliseHeadingsTo=4}}}
+
+                                    {{#isAnyTrue this hasPathParams hasHeaderParams hasBodyParam hasFormParams hasQueryParams}}
+                                    <h4 class="nhsd-t-heading-m">Request</h4>
+                                    {{/isAnyTrue}}
+
+                                    {{#hasPathParams}}
+                                        {{#with pathParams}}
+                                            {{>params_table headingLevel=5 headingText='Path parameters' headingClass='nhsd-t-heading-s'}}
```

Comment:
i wonder if the headingLevel and headingClass could be set as a scoped constant rather than hardcoding it for each params block?

maybe i'm trying to overoptimise lol

Reply:
I see your point: indeed, there is some repetition across invocations of various type _request_ parameters.

In 'normal' code I'd probably do just that, though, I think that in this case I prefer this repetition over a common constant. Purely subjectively, this version feels to me a bit more readable than adding a variable/constant somewhere higher up in the template.

Hopefully, given that all invocations of this template fit on a single screen and can be seen in one glance reduces the risk of the heading levels and classes drifting out of sync. I've now rearranged the order of the arguments in a way that, hopefully, reduces this risk even further, but aligning `headingLevel` and `headingClass` in the first two 'columns':
```
{{>params_table headingLevel=5 headingClass='nhsd-t-heading-s' headingText='Path parameters'}}
...
{{>params_table headingLevel=5 headingClass='nhsd-t-heading-s' headingText='Query parameters'}}
...
{{>params_table headingLevel=5 headingClass='nhsd-t-heading-s' headingText='Headers'}}
...
{{>params_table headingLevel=5 headingClass='nhsd-t-heading-s' headingText='Form parameters'}}
...
{{>params_table headingLevel=6 headingClass='nhsd-t-heading-xs' headingText='Headers' response=true}}
```